core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -1203,8 +1297,13 @@ private class BufferReleasingInputStream(
       block
     } catch {
       case e: IOException if detectCorruption =>
+        val diagnosisResponse = checkedInOpt.map { checkedIn =>
+          iterator.diagnoseCorruption(checkedIn, address, blockId)
+        }.orNull
```

Comment:
why we call `.orNull` here then pass `Some(diagnosisResponse)`? Shall we just keep `diagnosisResponse` here as an `Option`?